internal/metarepos/storage.go

```diff
@@ -341,11 +341,11 @@ func (ms *MetadataStorage) unregistableStorageNode(snID types.StorageNodeID) boo
 
 func (ms *MetadataStorage) unregisterStorageNode(snID types.StorageNodeID) error {
 	if ms.lookupStorageNode(snID) == nil {
-		return verrors.ErrNotExist
+		return status.Errorf(codes.NotFound, "sn %d", snID)
 	}
 
 	if !ms.unregistableStorageNode(snID) {
-		return verrors.ErrInvalidArgument
+		return status.Errorf(codes.InvalidArgument, "unregistable sn %d", snID)
```

Comment:
The client receiving the InvalidArgument may mistranslate this error. Could you take other error codes that are more fittable?

```go
	// InvalidArgument indicates client specified an invalid argument.
	// Note that this differs from FailedPrecondition. It indicates arguments
	// that are problematic regardless of the state of the system
	// (e.g., a malformed file name).
	//
	// This error code will not be generated by the gRPC framework.
	InvalidArgument Code = 3
```